pyOCD/tools/pyocd.py

```diff
@@ -69,207 +76,490 @@ def dumpHexData(data, startAddress=0, width=8):
+
+class PyOCDConsole(object):
+    PROMPT = '>>> '
+
+    def __init__(self, tool):
+        self.tool = tool
+        self.last_command = ''
+
+    def run(self):
+        try:
+            while True:
+                try:
+                    line = raw_input(self.PROMPT)
+                    line = line.strip()
+                    if line:
+                        self.process_command_line(line)
+                        self.last_command = line
+                    elif self.last_command:
+                        self.process_command(self.last_command)
+                except KeyboardInterrupt:
+                    print
+        except EOFError:
+            print
+
+    def process_command_line(self, line):
+        for cmd in line.split(';'):
+            self.process_command(cmd)
+
+    def process_command(self, cmd):
+        try:
+            args = cmd.split()
+            cmd = args[0].lower()
+            args = args[1:]
+
+            # Handle help.
+            if cmd in ['?', 'help']:
+                self.show_help(args)
+                return
+
+            # Handle register name as command.
+            if cmd in pyOCD.target.cortex_m.CORE_REGISTER:
+                self.tool.handle_reg([cmd])
+                return
+
+            # Check for valid command.
+            if cmd not in self.tool.command_list:
+                print "Error: unrecognized command '%s'" % cmd
+                return
+
+            # Run command.
+            handler = self.tool.command_list[cmd]
+            handler(args)
+        except ValueError:
+            print "Error: invalid argument"
+            traceback.print_exc()
+        except pyOCD.transport.transport.TransferError:
+            print "Error: transfer failed"
+        except ToolError, e:
+            print "Error:", e
+
+    def show_help(self, args):
+        if not args:
+            self.list_commands()
+
+    def list_commands(self):
+        cmds = sorted(self.tool.command_list.keys())
+        print "Commands:\n---------"
+        print '\n'.join(cmds)
 
 class PyOCDTool(object):
     def __init__(self):
-        # logging.basicConfig(level=logging.INFO)
-        pass
+        self.board = None
+        self.exitCode = 0
+        self.command_list = {
+                'list' :    self.handle_list,
+                'erase' :   self.handle_erase,
+                'unlock' :  self.handle_unlock,
+                'info' :    self.handle_info,
+                'i' :       self.handle_info,
+                'status' :  self.handle_status,
+                'stat' :    self.handle_status,
+                'reg' :     self.handle_reg,
+                'reset' :   self.handle_reset,
+                'read' :    self.handle_read8,
+                'read8' :   self.handle_read8,
+                'read16' :  self.handle_read16,
+                'read32' :  self.handle_read32,
+                'r' :       self.handle_read8,
+                'r8' :      self.handle_read8,
```

Comment:
Oops, that should've been obvious. I'll just remove the 'r8' and 'w8' aliases in favour of 'r' and 'w'.